tests/syft/core/io/address_test.py

```diff
@@ -159,6 +140,242 @@ def test_to_string() -> None:
+    # Test getter
+    device = SpecificLocation(id=UID())
+    address_with_device = Address(
+        network=SpecificLocation(id=UID()),
+        domain=SpecificLocation(id=UID()),
+        device=device,
+        vm=SpecificLocation(id=UID()),
+    )
+    # Test device getter
+    assert address_with_device.device == device
+
+    # Test device setter
+    an_id = UID(value=uuid.UUID(int=333779996850170035686993356951732753684))
+    new_device = SpecificLocation(id=an_id)
+    address_with_device.device = new_device
+    assert address_with_device.device == new_device
+
+    # Test domain_id getter
+    address_without_device = Address(
+        network=SpecificLocation(id=UID()),
+        domain=SpecificLocation(id=UID()),
+        vm=SpecificLocation(id=UID()),
+    )
+    assert address_with_device.device_id == an_id
+    assert address_without_device.device_id is None
+
+
+def test_vm_and_vm_id_property_methods() -> None:
+    """Unit test for Address.vm and Address.vm_id methods"""
+    # Test getter
+    vm = SpecificLocation(id=UID())
+    address_with_vm = Address(
+        network=SpecificLocation(id=UID()),
+        domain=SpecificLocation(id=UID()),
+        device=SpecificLocation(id=UID()),
+        vm=vm,
+    )
+    # Test device getter
+    assert address_with_vm.vm == vm
+
+    # Test device setter
+    an_id = UID(value=uuid.UUID(int=333779996850170035686993356951732753684))
+    new_vm = SpecificLocation(id=an_id)
+    address_with_vm.vm = new_vm
+    assert address_with_vm.vm == new_vm
+
+    # Test domain_id getter
+    address_without_vm = Address(
+        network=SpecificLocation(id=UID()),
+        domain=SpecificLocation(id=UID()),
+        device=SpecificLocation(id=UID()),
+    )
+    assert address_with_vm.vm_id == an_id
+    assert address_without_vm.vm_id is None
+
+
+def test_target_id_property_method_with_a_return() -> None:
+    """Unit test for Address.target_id method"""
+    network = SpecificLocation(id=UID())
+    domain = SpecificLocation(id=UID())
+    device = SpecificLocation(id=UID())
+    vm = SpecificLocation(id=UID())
+    address = Address(
+        network=network,
+        domain=domain,
+        device=device,
+        vm=vm,
+    )
+    assert address.target_id == vm
+    address.vm = None
+    assert address.target_id == device
+    address.device = None
+    assert address.target_id == domain
+    address.domain = None
+    assert address.target_id == network
+
+
+def test_addressables_property_method() -> None:
+    """Unit test for Address.target_emoji method"""
+    an_id = UID(value=uuid.UUID(int=333779996850170035686993356951732753684))
+    network = SpecificLocation(id=an_id)
+    domain = SpecificLocation(id=an_id)
+    device = SpecificLocation(id=an_id)
+    vm = SpecificLocation(id=an_id)
+    address = Address(
+        network=network,
+        domain=domain,
+        device=device,
+        vm=vm,
+    )
+    assert address.addressables == [vm, device, domain, network]
```

Comment:
Great job overall! I think this should be enough as a test, setting the others to None might not be testing something. I would leave only line 370.

Reply:
The coverage doesn't change but it is the only place I can see where we are testing that the internal state of `address.addressables` is affected by those properties. In fact we don't use that property anywhere, so maybe we should just delete it? 🤷🏼‍♀️ I would say delete or keep the additional tests, either way.

Reply:
Addressed. Thanks!